src/index.js

```diff
@@ -2,41 +2,86 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  const price = product.querySelector('.price span').innerHTML;
+  const quantity = product.querySelector('.quantity input').value;
+  let subtotal = price * quantity;
+  product.querySelector('.subtotal span').innerHTML = subtotal;
+  return subtotal;
 
   //... your code goes here
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+/*   const singleProduct = document.querySelector('.product');
+  updateSubtotal(singleProduct); */
   // end of test
 
   // ITERATION 2
   //... your code goes here
+  let allItems = document.getElementsByClassName('product');
+  let total = 0;
+  for ( let i = 0; i < allItems.length; i++) {
+    total += updateSubtotal(allItems[i])
+  }
+  
 
   // ITERATION 3
   //... your code goes here
+  let totalValue = document.getElementById('total-value');
+  totalValue.innerHTML = total;
+  return total;
+
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
+  let grandparentDiv = target.parentNode.parentNode.parentNode;
+  grandparentDiv.removeChild(target.parentNode.parentNode);
+  calculateAll();
   //... your code goes here
 }
 
 // ITERATION 5
 
 function createProduct() {
   //... your code goes here
+  let newProduct = document.createElement('tr');
```

Comment:
Another way of doing this is by instead of creating a `tr` element and then target that element to give it the class `product`, you could target the `tbody` and insert all the `html` inside:

```
function createProduct() {
  let newProductInputs = document.querySelectorAll(".create-product input");
  let newProductName = newProductInputs[0].value;
  let newProductPrice = newProductInputs[1].value;

  // Saving the newProduct html code in a variable
  let newProduct = `
  <tr class="product">
    <td class="name">
      <span>${newProductName}</span>
    </td>
    <td class="price">$<span>${newProductPrice}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</button>
    </td>
  </tr>`;

  // Targeting the table
  const tableBody = document.querySelector("tbody");
  // Adding the newProduct to the end of the table
  tableBody.insertAdjacentHTML("beforeend", newProduct);

  // Adding the removeProduct function to the remove button
  let removeBtn = document.querySelectorAll(".btn-remove");
  removeBtn.forEach((btn) => btn.addEventListener("click", removeProduct));
}
```